contributors/devel/api_changes.md

```diff
@@ -885,32 +885,30 @@ One possible place to do this is in the REST storage strategy's PrepareForCreate
       newFrobber := obj.(*api.Frobber)
       oldFrobber := old.(*api.Frobber)
     
-      if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) {
-        newFrobber.Spec.ConfigSource = nil
-        oldFrobber.Spec.ConfigSource = nil
+      if !utilfeature.DefaultFeatureGate.Enabled(features.Frobber2D) {
+        newFrobber.Width = nil
+        oldFrobber.Width = nil
       }
     }
     ```
 
-4. In validation, ensure the alpha field is not set if the feature gate is disabled:
+4. In validation, ignore the alpha field if the feature gate is disabled:
 
     ```go
     func ValidateFrobber(f *api.Frobber, fldPath *field.Path) field.ErrorList {
       ...
       if utilfeature.DefaultFeatureGate.Enabled(features.Frobber2D) {
         ... normal validation of width field ...
-      } else if f.Width != nil {
-        allErrs = append(allErrs, field.Forbidden(fldPath.Child("width"), "disabled by feature-gate"))
```

Comment:
This was sort of belt and suspenders to make sure alpha data isn't persisted, especially since the clearing of the field is not as automatic as we'd want yet. For fields in reused types (like PodTemplateSpec), we're currently counting on manual sweeping to catch all the handlers where we need to clear

Reply:
for example, this would help with the concern noted in https://github.com/kubernetes/kubernetes/pull/50924#issuecomment-325045360